src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -884,26 +910,33 @@ protected static boolean containsIndelLargerOrSmallerThan(final VariantContext v
     }
 
     /**
+     * Find the number of no-call genotypes
+     *
+     * @param vc the variant context
+     * @return number of filtered samples
+     */
+    private int numNoCallGenotypes(final VariantContext vc) {
+        return numFilteredGenotypes(vc, Genotype::isNoCall);
+    }
+
+    /**
      * Find the number of filtered samples
      *
-     * @param vc the variant rod VariantContext
+     * @param vc the variant context
      * @return number of filtered samples
      */
     private int numFilteredGenotypes(final VariantContext vc)  {
-        if (vc == null) {
-            return 0;
-        }
-
-        int numFiltered = 0;
-        // check if we find it in the variant rod
-        final GenotypesContext genotypes = vc.getGenotypes(samples);
-        for (final Genotype g : genotypes) {
-            if (g.isFiltered() && !g.getFilters().isEmpty()) {
-                numFiltered++;
-            }
-        }
+        return numFilteredGenotypes(vc, g -> g.isFiltered() && !g.getFilters().isEmpty());
```

Comment:
Is `g.isFiltered() && g.getFilters().isEmpty` a case that actually occurs?  That sounds like an invalid state of some sort.  I think we should probably patch htsjdk so that doesn't happen.